src/spec.c

```diff
@@ -2085,6 +2092,49 @@ static void Indexes_ScanProc(RedisModuleCtx *ctx, RedisModuleString *keyname, Re
   if (scanner->cancelled) {
     return;
   }
+
+  RedisModuleServerInfoData *memory_data = RedisModule_GetServerInfo(ctx, "memory");
```

Comment:
You can use `setMemoryInfo` and the globals `memoryLimit` and `used_memory` instead of reimplementing the info retrieval again